cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedWorkflowProgramRunner.java

```diff
@@ -114,7 +113,7 @@ protected ProgramController launch(Program program, ProgramOptions options,
       tempDir.mkdirs();
       try {
         launcherFile = File.createTempFile("launcher", ".jar", tempDir);
-        MapReduceContainerHelper.saveLauncher(hConf, launcherFile, extraClassPaths);
+        MapReduceContainerHelper.saveLauncher(hConf, launcherFile, Lists.newArrayList(extraClassPaths));
```

Comment:
You don't need to copy if you modify the `saveLauncher` method to take a `Iterable<String>` instead of `LIst<String>`